makefiles/tools/openocd-adapters/stlink.inc.mk

```diff
@@ -8,3 +11,11 @@ ifneq (,$(DEBUG_ADAPTER_ID))
   OPENOCD_ADAPTER_INIT += -c 'hla_serial $(DEBUG_ADAPTER_ID)'
 endif
 export OPENOCD_ADAPTER_INIT
+
+# call a 'reset halt' command before starting the debugger
+export OPENOCD_DBG_START_CMD = -c 'reset halt'
+
+# select OpenOCD configuration depending on CPU type
+ifeq (,$(OPENOCD_CONFIG))
```

Comment:
Follow up to setting to the default value in boards:

You could add another check if `$(RIOTBOARD)/$(BOARD)/dist/openocd.cfg` does not exist before using the st configuration.